tephra-core/src/main/java/co/cask/tephra/persist/HDFSTransactionLogReaderV2.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.tephra.persist;
+
+import co.cask.tephra.TxConstants;
+import com.google.common.primitives.Bytes;
+import org.apache.hadoop.io.DataOutputBuffer;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.SequenceFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayOutputStream;
+import java.io.DataOutputStream;
+import java.io.EOFException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+/**
+ * {@link TransactionLogReader} that can read v2 version of Transaction logs. The logs are expected to
+ * have a sequence of {@link TransactionEdit}s along with commit markers that indicates the size of the batch of
+ * {@link TransactionEdit}s that were synced together. If the expected number of {@link TransactionEdit}s are not
+ * present then that set of {@link TransactionEdit}s are discarded.
+ */
+public class HDFSTransactionLogReaderV2 extends AbstractHDFSLogReader {
+  private static final Logger LOG = LoggerFactory.getLogger(HDFSTransactionLogReaderV2.class);
+
+  private final Queue<TransactionEdit> transactionEdits;
+  private final LongWritable key;
+
+  public HDFSTransactionLogReaderV2(SequenceFile.Reader reader) {
+    super(reader);
+    this.transactionEdits = new ArrayDeque<>();
+    this.key = new LongWritable();
+  }
+
+  @Override
+  public TransactionEdit next() throws IOException {
+    return next(null);
+  }
+
+  @Override
+  public TransactionEdit next(TransactionEdit reuse) throws IOException {
```

Comment:
Since we don't allow reuse in `HDFSTransactionLogReaderV2` anymore, and we'll deprecate `HDFSTransactionLogReaderV1`, it is better we change the interface.